src/hazelweb/Update.re

```diff
@@ -74,174 +75,256 @@ let log_action = (action: ModelAction.t, _: State.t): unit => {
   };
 };
 
+let evaluate_and_schedule =
+    (state: State.t, ~schedule_action, model: Model.t): Model.t => {
+  /* Send evaluation request. */
+  let pushed = model |> State.evaluator_next(state);
+
+  /* Set evaluation to pending after short timeout. */
+  /* FIXME: This is problematic if evaluation finished in time, but UI hasn't
+   * updated before below action is scheduled. */
```

Comment:
UI updates should be finished before incr_dom pops the next action right?

Reply:
Could be? I'm not too familiar with incr_dom. That would make sense though.

Reply:
I feel like there could still be a case where the actions are pushed out of order.